Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -0,0 +1,288 @@
+	}
+	else if (UByteProperty* ByteProperty = Cast<UByteProperty>(Property))
+	{
+		Schema_AddUint32(Object, Id, (std::uint32_t)ByteProperty->GetPropertyValue(Data));
+	}
+	else if (UUInt16Property* UInt16Property = Cast<UUInt16Property>(Property))
+	{
+		Schema_AddUint32(Object, Id, (std::uint32_t)UInt16Property->GetPropertyValue(Data));
+	}
+	else if (UUInt32Property* UInt32Property = Cast<UUInt32Property>(Property))
+	{
+		Schema_AddUint32(Object, Id, UInt32Property->GetPropertyValue(Data));
+	}
+	else if (UUInt64Property* UInt64Property = Cast<UUInt64Property>(Property))
+	{
+		Schema_AddUint64(Object, Id, UInt64Property->GetPropertyValue(Data));
+	}
+	else if (UObjectPropertyBase* ObjectProperty = Cast<UObjectPropertyBase>(Property))
+	{
+		UnrealObjectRef ObjectRef = SpatialConstants::NULL_OBJECT_REF;
+
+		UObject* ObjectValue = ObjectProperty->GetObjectPropertyValue(Data);
+		if (ObjectValue != nullptr)
+		{
+			FNetworkGUID NetGUID = PackageMap->GetNetGUIDFromObject(ObjectValue);
+			if (!NetGUID.IsValid())
+			{
+				if (ObjectValue->IsFullNameStableForNetworking())
+				{
+					NetGUID = PackageMap->ResolveStablyNamedObject(ObjectValue);
+				}
+			}
+			ObjectRef = UnrealObjectRef(PackageMap->GetUnrealObjectRefFromNetGUID(NetGUID));
+			if (ObjectRef == SpatialConstants::UNRESOLVED_OBJECT_REF)
+			{
+				// A legal static object reference should never be unresolved.
+				check(!ObjectValue->IsFullNameStableForNetworking());
+				UnresolvedObjects.Add(ObjectValue);
+				ObjectRef = SpatialConstants::NULL_OBJECT_REF;
+			}
+		}
+
+		Schema_AddObjectRef(Object, Id, ObjectRef);
+	}
+	else if (UNameProperty* NameProperty = Cast<UNameProperty>(Property))
+	{
+		Schema_AddString(Object, Id, NameProperty->GetPropertyValue(Data).ToString());
+	}
+	else if (UStrProperty* StrProperty = Cast<UStrProperty>(Property))
+	{
+		Schema_AddString(Object, Id, StrProperty->GetPropertyValue(Data));
+	}
+	else if (UTextProperty* TextProperty = Cast<UTextProperty>(Property))
+	{
+		Schema_AddString(Object, Id, TextProperty->GetPropertyValue(Data).ToString());
+	}
+	else if (UArrayProperty* ArrayProperty = Cast<UArrayProperty>(Property))
+	{
+		FScriptArrayHelper ArrayHelper(ArrayProperty, Data);
+		for (int i = 0; i < ArrayHelper.Num(); i++)
+		{
+			AddProperty(Object, Id, ArrayProperty->Inner, ArrayHelper.GetRawPtr(i), UnresolvedObjects, ClearedIds);
+		}
+
+		if (ArrayHelper.Num() == 0 && ClearedIds)
+		{
+			ClearedIds->Add(Id);
+		}
+	}
+	else if (UEnumProperty* EnumProperty = Cast<UEnumProperty>(Property))
+	{
+		if (EnumProperty->ElementSize < 4)
+		{
+			Schema_AddUint32(Object, Id, (std::uint32_t)EnumProperty->GetUnderlyingProperty()->GetUnsignedIntPropertyValue(Data));
+		}
+		else
+		{
+			AddProperty(Object, Id, EnumProperty->GetUnderlyingProperty(), Data, UnresolvedObjects, ClearedIds);
+		}
+	}
+	else
+	{
+		checkNoEntry();
+		Schema_AddString(Object, Id, TEXT("Unknown Field"));
```

Comment:
Don't write schema data that doesn't match the schema that the runtime understands. This can probably cause various hard to debug issues. Best to print a warning here, or add a message to `checkNoEntry`.

Reply:
Also, this check should probably match the checks in `ApplyProperty` and `GetPropertyCount`.